cluster/gce/gci/configure-helper.sh

```diff
@@ -173,8 +173,9 @@ function config-ip-firewall {
   # node because we don't expect the daemonset to run on this node.
   if [[ "${ENABLE_METADATA_CONCEALMENT:-}" == "true" ]] && [[ ! "${METADATA_CONCEALMENT_NO_FIREWALL:-}" == "true" ]]; then
     echo "Add rule for metadata concealment"
-    iptables -w -t nat -I PREROUTING -p tcp ! -i eth0 -d "${METADATA_SERVER_IP}" --dport 80 -m comment --comment "metadata-concealment: bridge traffic to metadata server goes to metadata proxy" -j REDIRECT --to-ports 988
-    iptables -w -t nat -I PREROUTING -p tcp ! -i eth0 -d "${METADATA_SERVER_IP}" --dport 8080 -m comment --comment "metadata-concealment: bridge traffic to metadata server goes to metadata proxy" -j REDIRECT --to-ports 987
+    ip route add table local local 169.254.169.252 dev lo proto kernel scope host src 169.254.169.252
```

Comment:
Can we just add this address to eth0?
```
ip addr add dev eth0 169.254.169.252/32
```
It's a lot more obvious then hiding something in the local route table.

Reply:
done

Reply:
you can add it to the loopback interface, any special reason to use eth0? 
```
$ sudo ip addr add 169.254.169.252 dev lo
$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet 169.254.169.252/32 scope global lo
```
is not a good idea to have this link local addresses in public interfaces ... but is not wrong though

Reply:
thanks for pointing it out. i meant to add it to lo. done.

Reply:
Apparently route_localnet only cares about 127/8, not that the requests target loopback?

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/ipv4/route.c?h=v5.11#n1710

If this works in testing, lo seems preferable to eth0.

Reply:
yeah, localnet is hack to solve the martian problem with localhost addresses 
```
route_localnet - BOOLEAN
    Do not consider loopback addresses as martian source or destination
    while routing. This enables the use of 127/8 for local routing purposes.
    default FALSE
```

this is just another ip ... in some other interface